tests/test_views_objects_permissions.py

```diff
@@ -236,3 +235,41 @@ def test_safe_creation_with_post_returns_403_if_only_allowed_to_read(self):
         self.app.post_json('/buckets/read/groups',
                            {'data': {'id': 'child', 'members': []}},
                            headers=self.bob_headers_safe_creation, status=403)
+
+
+class ParentMetadataTest(PermissionsTest):
+    def setUp(self):
+        self.app.put_json('/buckets/beer',
+                          {'permissions': {'collection:create': [self.bob_principal]}},
+                          headers=self.headers)
+        self.app.put_json('/buckets/beer/collections/barley',
+                          {'permissions': {'record:create': [self.alice_principal]}},
+                          headers=self.bob_headers)
+
+    def test_parent_metadata_can_be_read_if_allowed_to_create_child(self):
+        self.app.get('/buckets/beer', headers=self.bob_headers)
+        self.app.get('/buckets/beer/collections/barley', headers=self.alice_headers)
+
+    def test_parent_metadata_cannot_be_read_if_not_allowed_to_create_child(self):
+        self.app.get('/buckets/beer',
+                     headers=get_user_headers('jean:paul'),
+                     status=403)
+        self.app.get('/buckets/beer/collections/barley',
+                     headers=get_user_headers('mahmud:hatim'),
+                     status=403)
+
+    def test_list_can_be_obtained_if_allowed_to_create(self):
+        resp = self.app.get('/buckets/beer/collections', headers=self.bob_headers)
+        self.assertEqual(len(resp.json['data']), 1)
+        self.assertEqual(resp.json['data'][0]['id'], 'barley')
```

Comment:
the `barley` collection was created by Bob (and Alice is allowed to create records). But you're right, the current version of the patch is not correct. I added a failing test that shows that users should only see their own records in this particular case.